examples/decoding/cohy_decomposition.py

```diff
@@ -0,0 +1,557 @@
+# As a side note, it is important to consider that a multivariate approach may be as
+# fast or even faster than a bivariate approach, depending on the number of connections
+# and degree of rank subspace projection being performed.
+
+# %%
+
+# Time transforming data of each epoch iteratively
+start = time.time()
+for epoch in epochs[N_EPOCHS // 2 :]:
+    spectral_connectivity_epochs(
+        np.expand_dims(epoch, axis=0),
+        method="coh",
+        indices=seed_target_indices(indices[0], indices[1]),
+        fmin=5,
+        fmax=35,
+        sfreq=epochs.info["sfreq"],
+    )
+duration = (time.time() - start) * 1000
+
+# %%
+
+print(f"Time to compute connectivity: {duration:.0f} ms")
+
+print(
+    "\nTime to compute connectivity per epoch (window): ",
+    f"{duration/(N_EPOCHS//2):.0f} ms",
+)
+
+########################################################################################
+# Case 2: Fitting to and transforming the same data
+# -------------------------------------------------
+#
+# As mentioned above, the decoding module classes can also be used to transform the same
+# data the filters are fit to. This is a similar process to that of the
+# ``spectral_connectivity_...()`` functions, but with the increased efficiency of
+# fitting filters to a single frequency band as opposed to each frequency bin.
+#
+# To demonstrate this approach, we will load some example MEG data and divide it into
+# two-second-long epochs. We designate the left hemisphere sensors as the seeds and the
+# right hemisphere sensors as the targets. Since this is sensor-space data, we will use
+# the MIC method to analyse connectivity given its resilience to zero time-lag
+# interactions (See :doc:`../compare_coherency_methods` for more information).
+
+# %%
+
+raw = mne.io.read_raw_ctf(data_path() / "SubjectCMC.ds")
+raw.pick("mag")
+raw.crop(50.0, 110.0).load_data()
+raw.notch_filter(50)
+raw.resample(100)
+
+epochs = make_fixed_length_epochs(raw, duration=2.0).load_data()
+
+# left hemisphere sensors
+seeds = [idx for idx, ch_info in enumerate(epochs.info["chs"]) if ch_info["loc"][0] < 0]
+# right hemisphere sensors
+targets = [
+    idx for idx, ch_info in enumerate(epochs.info["chs"]) if ch_info["loc"][0] > 0
+]
+
+########################################################################################
+# There are two equivalent options for fitting and transforming the same data: 1)
+# passing the data to the :meth:`~mne_connectivity.decoding.CoherencyDecomposition.fit`
+# and :meth:`~mne_connectivity.decoding.CoherencyDecomposition.transform` methods
+# sequentially; or 2) using the combined
+# :meth:`~mne_connectivity.decoding.CoherencyDecomposition.fit_transform` method.
+#
+# We use the latter approach below, fitting the filters to the 15-20 Hz band and using
+# the ``"imcoh"`` method in the call to the ``spectral_connectivity_...()`` functions.
+# Plotting the results, we see a peak in connectivity at 15-20 Hz.
```

Comment:
something is confusing me here. The first example faked some data with simulated connectivity in the 15-20Hz band, fit filters from 5-35 Hz, and recovered the fact that connectivity was higher in the 15-20Hz band.  But in this example, you load *real data* and fit filters on the 15-20Hz band, and transform the data using those filters... so IIUC the plot for case 2 basically just shows that the filter worked.  Do I have that right?  

If so, what I find confusing about it is how similar the plots are styled, as if they're showing basically the same thing on fake vs real data. But case 1 is showing "we can recover actual (simulated) connectivity and we don't see spurious connectivity where it doesn't exist", while case 2 is showing "the spatial filters actually capture connectivity in the band we asked for, and (mostly) suppress connectivity in frequencies outside that band".  I'm not totally sure how best to improve the tutorial to overcome this confusion; maybe first you can confirm that I'm actually understanding / interpreting correctly, then we can talk about how to improve it.

Reply:
> case 1 is showing "we can recover actual (simulated) connectivity and we don't see spurious connectivity where it doesn't exist", while case 2 is showing "the spatial filters actually capture connectivity in the band we asked for, and (mostly) suppress connectivity in frequencies outside that band"

So case 1 is just about showing that you can fit filters to one piece of data and apply it to another (not necessarily about showing there is no spurious connectivity), and case 2 is about fitting filters to and transforming the same piece of data (not necessarily that the filters are capturing connectivity in a given band and suppressing it elsewhere). For either/both of these I could have used real/simulated data.

For case 1 I looked at using the fieldtrip_cmc data for this, but found it difficult to identify a connectivity component that was present across multiple parts of the recording duration (which is necessary if you want to fit the filters to one set of epochs and apply them to another). I'm sure I could have explored this more, but in the interest of time I settled on using simulated data where I know there is a consistent connectivity component throughout the whole length of data.

For case 2 I could have used simulated data again, but I always think these tutorials look more convincing with some real data examples, and for this case since we are fitting to and transforming the same data, identifying a connectivity component which is consistent across multiple parts of the recording is not necessary.

<br>

> The first example faked some data with simulated connectivity in the 15-20Hz band, fit filters from 5-35 Hz, and recovered the fact that connectivity was higher in the 15-20Hz band

For case 1, I start by showing the connectivity of the data that our simulated data indeed involves an interaction at 15-20 Hz. I compute connectivity using the `spectral_connectivity_epochs()` function where filters are fit to each frequency bin.

I then use the new decomposition class to fit filters to **only the 15-20 Hz range** which are trained on the first 30 epochs and applied to the last 30 epochs. Alongside this I show the results of the `spectral_connectivity_epochs()` function for the last 30 epochs (i.e. fitting and transforming these epochs, with filters fit to each frequency bin).

**This is to demonstrate that the filters trained on one piece of data can extract this same connectivity component from a new piece of data**. The point is not to show that the filters do not pick up spurious connectivity.

<br>

> you load real data and fit filters on the 15-20Hz band, and transform the data using those filters... so IIUC the plot for case 2 basically just shows that the filter worked.

In case 2 I show that you can also use the decomposition class to fit and transform the same piece of data, similar to what is happening in the `spectral_connectivity_...` functions (except with fitting filters to a single band, not each frequency bin).

In the same way as for case 1 I am still only fitting filters to 15-20 Hz using the decomposition class. The first plot where the results of fitting & transforming the same data using the decomposition classes is shown alone is just to demonstrate that there is connectivity at 15-20 Hz in the data.

Then I compute connectivity using the `spectral_connectivity_epochs()` function (where filters are fit to each frequency bin) to show that fitting and transforming the same piece of data using the decomposition class gives a near-identical result to the `spectral_connectivity_...` approach, at least for the 15-20 Hz range where the decomposition class fit the filters.

**The point is to demonstrate that fitting to and transforming the same piece of data with the decomposition class is a valid use-case**. The point is not that connectivity outside the given frequency band is suppressed (it's not actively suppressed, the filters just don't care about optimising this connectivity). It does show that "the spatial filters actually capture connectivity in the band we asked for", but this is also shown in case 1.

<br>

Does this make it any clearer about what each example is trying to show? These are the key points:
- case 1 is just about showing that you can fit filters to one piece of data and apply it to another
- case 2 is about fitting filters to and transforming the same piece of data

Maybe having a final summary after case 1 & 2 helps get this across.

Reply:
> This is to demonstrate that the filters trained on one piece of data can extract this same connectivity component from a new piece of data. The point is not to show that the filters do not pick up spurious connectivity.

But perhaps the tutorial can be changed to include the point that the filters do not pick up spurious connectivity. This is an important argument in support of the statement "can extract the same connectivity component".

At any rate, I feel this is something we want users to think about when they make decisions regarding what parts of the signal to fit the filters on and what parts to apply the filters to. The tutorial can help set them on the right path.